test_http.py

```diff
@@ -0,0 +1,124 @@
+import echo_client as ec
+import echo_server as es
+import subprocess
+import pytest
+import email.utils
+
+
+# @pytest.fixture(scope='module')
+# def server(request):
+#     """set up and tear down a server"""
+
+#     process = subprocess.Popen('./echo_server.py', shell=True)
+
+#     def cleanup():
+#         process.kill()
+
+#     request.addfinalizer(cleanup)
+
+#     return process
+
+
+def test_response_ok():
+    ''' Test ok response message. '''
+    first_line = 'HTTP/1.1 200 OK'
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    content_header = 'Content-Type: text/plain'
+    body = '200 OK'
+    response_list = [first_line, timestamp, content_header, ' ', body, '\r\n']
+    assert es.response_ok() == '\r\n'.join(response_list)
+
+
+def test_response_error():
+    ''' Test error response message. '''
+    error_code = '404'
+    error_message = 'Not Found'
+    first_line = 'HTTP/1.1 {} {}'.format(error_code, error_message)
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    content_header = 'Content-Type: text/plain'
+    body = '{} {}'.format(error_code, error_message)
+    response_list = [first_line, timestamp, content_header, ' ', body, '\r\n']
+    assert es.response_error(
+        error_code, error_message) == '\r\n'.join(response_list)
```

Comment:
It's probably enough to verify that when you call the response_error function passing in the code and message that the thing you get back is a valid HTTP response that contains the code and message on the first line.  Could you abstract the question "is a valid HTTP response" to a function?